src/oscar/apps/catalogue/abstract_models.py

```diff
@@ -154,6 +166,9 @@ def get_ancestors_and_self(self):
         if you don't want to include the category itself. It's a separate
         function as it's commonly used in templates.
         """
+        if self.is_root():
+            return [self]
```

Comment:
This guard removed 255 completely useless queries in a site we made, have a look at this:

![image](https://user-images.githubusercontent.com/52858/50376184-362eff00-060a-11e9-83ab-8cc8b8645b2a.png)

Reply:
Looks like something that really needs to be fixed upstream in Treebeard - we shouldn't have to do this. Will add an issue there.